orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/DeployCloudFormationStage.java

```diff
@@ -36,5 +38,8 @@ public void taskGraph(@Nonnull Stage stage, @Nonnull TaskNode.Builder builder) {
         .withTask("monitorCloudFormation", MonitorKatoTask.class)
         .withTask("forceRefreshCache", CloudFormationForceCacheRefreshTask.class)
         .withTask("waitForCloudFormationCompletion", WaitForCloudFormationCompletionTask.class);
+    if ((boolean) Optional.ofNullable(stage.getContext().get("isChangeSet")).orElse(false)) {
+      builder.withTask("deleteCloudFormationChangeSet", DeleteCloudFormationChangeSetTask.class);
```

Comment:
do you need to have some sort of `waitForCompletion` / `monitorKatoTask` after the deletion of the changeset?

Reply:
Added the monitor kato task. I'd like to keep it simple for now (no retries). I was looking at the saga backed operations, but this looks like a bigger change and understanding better how it works.